app/models/agents/website_agent.rb

```diff
@@ -106,24 +106,41 @@ class WebsiteAgent < Agent
 
       Set `http_success_codes` to an array of status codes (e.g., `[404, 422]`) to treat HTTP response codes beyond 200 as successes.
 
+      The `interpolate` option is the way to format extracted data, and add meta information to each resulted payload.  Its value must be a hash, whose key-value pairs are interpolated after extraction and added to each event payload.  e.g.:
+
+          "interpolate": {
```

Comment:
I'm not sure.  I named it `interpolate` because it should correspond to `extract`, and other agents already have the same functionality under different names: `template`, `instructions`, etc.

I need this in WebsiteAgent because it does not provide templating for what to output, but only for what to extract.  If you want to output a respond header value, there's no straightforward way to do that, but an ugly workaround like this:
```
"extract": {
  "url": { "css": "#comic img", "value": "@src" },
  "title": { "css": "#comic img", "value": "@title" },
  "body_text": { "css": "div.main", "value": ".//text()" },
  // CSS selector only for not getting an `uneven number of matches` error,
  // and the `to_xpath` hack
  "status": { "css": "div.main", "value": "{{ _request_.status | to_xpath }}" }
}
```
But this is only applicable to html/xml; there's no equivalent hack for the json and text extraction types.